### PR TITLE
Replace User-based contacts with Account-based contacts

### PR DESCRIPTION
**Pull request by hawx**

### CODE REVIEW

File: source/contacts/ContactService.cs [hunk 1]
```diff
@@ -1,131 +1,113 @@
...
+        /// </summary>
+        /// <param name="contact">A com.esendex.sdk.contacts.Contact instance that contains the contact.</param>
+        /// <returns>A com.esendex.sdk.contacts.Contact instance that contains the contact with an Id assigned.</returns>        
+        /// <exception cref="System.ArgumentException"></exception>
+        /// <exception cref="System.Net.WebException"></exception>        
+        public Contact CreateContact(Contact contact)
+        {
+            var requestXml = Serialiser.Serialise(contact);
+
+            Console.WriteLine(requestXml);
```
Comment (Codesleuth) [inline source/contacts/ContactService.cs]: Huh?

File: test/rest/RestClientTests.cs [hunk 2]
```diff
@@ -1,348 +1,368 @@
...
+            DeleteRestResourceContext deleteResourceContext = new DeleteRestResourceContext();
+
+            HttpRequest httpRequest = new HttpRequest()
+            {
+                HttpMethod = HttpMethod.DELETE,
+                ResourcePath = deleteResourceContext.ResourcePath,
+                ContentType = "text/plain"
+            };
+
+            RestResponse expectedResponse = new RestResponse()
```
Comment (Codesleuth) [inline test/rest/RestClientTests.cs]: Unused - but this isn't new? I wonder if we can do a ReSharper solution format and remove these redundancies...?

File: test/rest/RestClientTests.cs [hunk 3]
```diff
@@ -1,348 +1,368 @@
...
+            HttpRequest httpRequest = new HttpRequest()
+            {
+                HttpMethod = HttpMethod.PUT,
+                ResourcePath = putResourceContext.ResourcePath,
+                Content = putResourceContext.Content,
+                ContentType = "application/xml",
+                ContentEncoding = Encoding.UTF8
+            };
+
+            RestResponse expectedResponse = new RestResponse()
```
Comment (Codesleuth) [inline test/rest/RestClientTests.cs]: Also unused. See previous comment. :panda_face:

File: source/rest/RestResource.cs [hunk 4]
```diff
@@ -4,13 +4,15 @@ namespace com.esendex.sdk.rest
...
+        public abstract string ResourceVersion { get; }
 
         public string Content { get; protected set; }
-
         public string ResourcePath { get; protected set; }
+        public string Version { get; protected set; }
 
         public RestResource()
         {
+            Version = ResourceVersion;
```
Comment (Codesleuth) [inline source/rest/RestResource.cs]: What's this for?

File: source/ServiceBase.cs [hunk 5]
```diff
@@ -52,7 +50,7 @@ internal RestResponse MakeRequest(HttpMethod method, RestResource resource)
                 case HttpMethod.DELETE:
                     return RestClient.Delete(resource);
                 default:
-                    throw new ArgumentException(string.Format("An invalid HttpMethod was supplied for this type of resource."), "method");
+                    throw new ArgumentException("An invalid HttpMethod was supplied for this type of resource.", "method");
```
Comment (Codesleuth) [inline source/ServiceBase.cs]: Hah, what was this even doing?

File: source/contacts/ContactService.cs (hunk 1 above)
Comment (hawx) [inline source/contacts/ContactService.cs]: Ha

File: source/rest/RestResource.cs (hunk 4 above)
Comment (hawx) [inline source/rest/RestResource.cs]: I was just copying the way that ResourcePath is set to the value of ResourceName. I hate it, but didn't really feel like changing how all this works.

File: source/ServiceBase.cs (hunk 5 above)
Comment (hawx) [inline source/ServiceBase.cs]: `string.Format(A) => A`

**Pull request merged**